File: backend/api/app/domain/use_cases/auth/get_token.py
from core.use_cases.use_case import UseCase
from app.domain.repositories.auth import BaseRepository
from core.common.equatable import Equatable
from core.common.either import Either
from core.errors.failure import Failure
from app.domain.entities.auth import Auth, AuthEntity

class Params(Equatable):
    def __init__(self, auth: AuthEntity) -> None:
        self.auth = auth


class CreateChat(UseCase[Auth]):
    def __init__(self, repository: BaseRepository):
        self.repository = repository
    
    async def __call__(self, params: Params) -> Either[Failure, Auth]:
        return await self.repository.get_auth(params.auth)

File: mobile/lib/features/architect/domains/use_cases/auth/get_token.dart
import 'package:dartz/dartz.dart';
import 'package:equatable/equatable.dart';

import '../../../../../core/errors/failure.dart';
import '../../../../../core/use_cases/usecase.dart';
import '../../entities/auth.dart';
import '../../repositories/auth.dart';

class GetToken extends UseCase<Auth, Params> {
  final AuthRepository repository;

  GetToken(this.repository);

  @override
  Future<Either<Failure, Auth>> call(Params params) async {
    return await repository.getAuth(
      email: params.email,
      password: params.password,
    );
  }
}

class Params extends Equatable {
  final String email;
  final String password;

  const Params({required this.email, required this.password});

  @override
  List<Object?> get props => [email, password];
}
